graphene_sqlalchemy/filters.py

```diff
@@ -0,0 +1,463 @@
+import re
+from typing import Any, Dict, List, Tuple, Type, TypeVar, Union
+
+from sqlalchemy import and_, not_, or_
+from sqlalchemy.orm import Query, aliased  # , selectinload
+
+import graphene
+from graphene.types.inputobjecttype import (
+    InputObjectTypeContainer,
+    InputObjectTypeOptions,
+)
+from graphene_sqlalchemy.utils import is_list
+
+BaseTypeFilterSelf = TypeVar(
+    "BaseTypeFilterSelf", Dict[str, Any], InputObjectTypeContainer
+)
+
+
+def _get_functions_by_regex(
+    regex: str, subtract_regex: str, class_: Type
+) -> List[Tuple[str, Dict[str, Any]]]:
+    function_regex = re.compile(regex)
+
+    matching_functions = []
+
+    # Search the entire class for functions matching the filter regex
+    for fn in dir(class_):
+        func_attr = getattr(class_, fn)
+        # Check if attribute is a function
+        if callable(func_attr) and function_regex.match(fn):
+            # add function and attribute name to the list
+            matching_functions.append(
+                (re.sub(subtract_regex, "", fn), func_attr.__annotations__)
+            )
+    return matching_functions
+
+
+class BaseTypeFilter(graphene.InputObjectType):
+    @classmethod
+    def __init_subclass_with_meta__(
+        cls, filter_fields=None, model=None, _meta=None, **options
+    ):
+        from graphene_sqlalchemy.converter import convert_sqlalchemy_type
+
+        # Init meta options class if it doesn't exist already
+        if not _meta:
+            _meta = InputObjectTypeOptions(cls)
+
+        logic_functions = _get_functions_by_regex(".+_logic$", "_logic$", cls)
+
+        new_filter_fields = {}
+        print(f"Generating Filter for {cls.__name__} with model {model} ")
+        # Generate Graphene Fields from the filter functions based on type hints
+        for field_name, _annotations in logic_functions:
+            assert (
+                "val" in _annotations
+            ), "Each filter method must have a value field with valid type annotations"
+            # If type is generic, replace with actual type of filter class
+
+            replace_type_vars = {BaseTypeFilterSelf: cls}
+            field_type = convert_sqlalchemy_type(
+                _annotations.get("val", str), replace_type_vars=replace_type_vars
```

Comment:
I investigated the issue I am experiencing where all the and/or filters turn out to be of type "[String]".
So far I was unable to reproduce this outside of our production setup but I have narrowed it down to this line, where the annotation["val"] turns out to be typing.List[String] for and/or. I am still trying to create a reproducible test, but maybe this info already sparks a thought that helps @sabard or @erikwrede

Reply:
Got it! As soon as you register custom types someting in the registry gets messed up and will pick the wrong type for and/or.

This will result in and/or becoming [String]:

```python
import sqlalchemy.types
from graphene_sqlalchemy.converter import convert_sqlalchemy_type
from graphene import String

@convert_sqlalchemy_type.register(
    sqlalchemy.types.PickleType
)
def convert_column_to_string(*args, **kwargs):
    return String
```

Too late now but hopefully this helps a bit more. Will try to setup a test sometime soon.